cmd/bucket-targets.go

```diff
@@ -272,17 +272,20 @@ func (sys *BucketTargetSys) UpdateAllTargets(bucket string, tgts *madmin.BucketT
 	}
 	sys.Lock()
 	defer sys.Unlock()
-	if tgts == nil || tgts.Empty() {
-		// remove target and arn association
-		if tgts, ok := sys.targetsMap[bucket]; ok {
-			for _, t := range tgts {
-				if tgt, ok := sys.arnRemotesMap[t.Arn]; ok && tgt.healthCancelFn != nil {
-					tgt.healthCancelFn()
-				}
-				delete(sys.arnRemotesMap, t.Arn)
+
+	// Remove existingtarget and arn association
+	if tgts, ok := sys.targetsMap[bucket]; ok {
+		for _, t := range tgts {
+			if tgt, ok := sys.arnRemotesMap[t.Arn]; ok && tgt.healthCancelFn != nil {
+				tgt.healthCancelFn()
 			}
+			delete(sys.arnRemotesMap, t.Arn)
 		}
 		delete(sys.targetsMap, bucket)
+	}
+
+	// No need for more if not adding anything
+	if tgts == nil || tgts.Empty() {
```

Comment:
[Optional] With the simplification brought about in this PR we can remove the following [block](https://github.com/minio/minio/pull/15167/files#diff-4cb845cd18a0f200b38b18c6ae381eb5a9572e944482e62b18430b6ee968fe3fR293).
```go
if len(tgts.Targets) > 0 {
    sys.targetsMap[bucket] = tgts.Targets
}
```

Reply:
I agree. Didn't want to accidentally break something, so kept it to a minimum.